src/consent.ts

```diff
@@ -258,176 +264,233 @@ export default function Consent(this: IConsent, mpInstance: MParticleWebSDK) {
         },
     };
 
+    /**
+     * Invoke these methods on a consent state object.
+     * <p>
+     * Usage: var consent = mParticle.Consent.createConsentState()
+     * <br>
+     * consent.setGDPRCoonsentState()
+     *
+     * @class Consent
+     */
     this.createConsentState = function(
-        this: ConsentState,
-        consentState?: ConsentState
-    ): ConsentState {
-        var gdpr = {};
-        var ccpa = {};
+        this: ConsentState
+        // TODO: Based on usage, this should not take any params
+        // consentState?: ConsentState
+    ): ConsentStateOBJ {
+        // if (consentState) {
+        //     setGDPRConsentState(consentState.getGDPRConsentState());
+        //     setCCPAConsentState(consentState.getCCPAConsentState());
+        // }
+
+        return new ConsentStateOBJ();
+    };
+}
 
-        function canonicalizeForDeduplication(purpose: string): string {
-            if (typeof purpose !== 'string') {
-                return null;
-            }
-            var trimmedPurpose = purpose.trim();
-            if (!trimmedPurpose.length) {
-                return null;
-            }
-            return trimmedPurpose.toLowerCase();
-        }
+// TODO: Fix collisison between this and ConsentState
+class ConsentStateOBJ {
```

Comment:
is this like a `ConsentStateModel` and the other is more of a `ConsentStateFactory`??